### PR TITLE
Parsing mentions using moera commons and not regexes

### PR DESCRIPTION
**Pull request by eliduvid**

### CODE REVIEW

File: src/main/java/org/moera/node/text/markdown/mention/internal/MentionsInlineParserExtension.java [hunk 1]
```diff
@@ -29,26 +29,35 @@ public void finalizeBlock(@NotNull InlineParser inlineParser) {
...
+        final int startingIndex = inlineParser.getIndex();
+        boolean isPossible = startingIndex == 0;
         if (!isPossible) {
-            char c = inlineParser.getInput().charAt(index - 1);
+            char c = inlineParser.getInput().charAt(startingIndex - 1);
             if (!Character.isUnicodeIdentifierPart(c) && c != '-' && c != '.') {
                 isPossible = true;
             }
         }
         if (isPossible) {
```
Comment (smelamud) [inline src/main/java/org/moera/node/text/markdown/mention/internal/MentionsInlineParserExtension.java:40]: I just though that adding early return here would make the code even simpler.

**Review by smelamud — CHANGES_REQUESTED** [review]

File: src/test/java/org/moera/node/text/markdown/mention/MentionsTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                "<a " +
+                "href=\"/moera/gotoname?name=-%3D%21%21.%3F%3Fweird**UsErNamEE%2B--\" " +
+                "data-nodename=\"-=!!.??weird**UsErNamEE+--\">" +
+                "@-=!!.??weird**UsErNamEE+--" +
+                "</a>" +
+                " indeed</p>\n";
+        assertRendering("very @-=!!.??weird**UsErNamEE+-- indeed", expectedResult);
+    }
+
+}
```
Comment (smelamud) [inline src/test/java/org/moera/node/text/markdown/mention/MentionsTest.java]: Newline.

File: src/main/java/org/moera/node/text/markdown/mention/internal/MentionsNodeRenderer.java [hunk 3]
```diff
@@ -42,7 +42,7 @@ private void render(MentionNode node, NodeRendererContext context, HtmlWriter ht
                 try {
                     RegisteredNameDetails details = options.namingCache.get(name);
                     name = details.getNodeName() != null ? details.getNodeName() : name;
-                } catch (NamingNotAvailableException e) {
+                } catch (NamingNotAvailableException|NullPointerException ignore) {
```
Comment (smelamud) [inline src/main/java/org/moera/node/text/markdown/mention/internal/MentionsNodeRenderer.java:45]: Catching `NullPointerException` is too radical. If you know that `options.namingCache` may be `null`, just check this.

File: src/main/java/org/moera/node/text/markdown/mention/MentionsExtension.java [hunk 4]
```diff
@@ -13,7 +13,7 @@
 
     public static final DataKey<NamingCache> NAMING_CACHE = new DataKey<>("NAMING_CACHE", (NamingCache) null);
 
-    private MentionsExtension() {
+    public MentionsExtension() {
```
Comment (smelamud) [inline src/main/java/org/moera/node/text/markdown/mention/MentionsExtension.java]: Do you use the constructor anywhere? I see only calls to `create()`. Anyway, if you need it for tests only, package access would be enough.

**Review by smelamud — CHANGES_REQUESTED** [review]

File: src/main/java/org/moera/node/text/markdown/mention/internal/MentionsInlineParserExtension.java [hunk 5]
```diff
@@ -30,36 +30,34 @@ public void finalizeBlock(@NotNull InlineParser inlineParser) {
     @Override
     public boolean parse(LightInlineParser inlineParser) {
         final int startingIndex = inlineParser.getIndex();
-        boolean isPossible = startingIndex == 0;
-        if (!isPossible) {
+        if (!(startingIndex == 0)) {
```
[the remark this answers is not in the document]
Comment (eliduvid) [reply src/main/java/org/moera/node/text/markdown/mention/internal/MentionsInlineParserExtension.java:33]: 🤦‍♂️

**Pull request merged**